### DISCUSSION TITLE
Restructure tests

### DISCUSSION THREAD

**Discussion by heitorlessa** [aws-powertools/powertools-lambda-python#331] [Maintainers discussions]
**Problem**: As more utilities were added such as `Parser` and `Event source data classes`, we have duplicated fixtures along with helper functions to retrieve event samples.

We should restructure our tests to make better use of `conftest`, and optionally clean up any test names/similar tests

**Comment by michaelbrewer** [comment]
@heitorlessa i do like the idea of having a directory of example events json for the fixtures (but also in general uses)

**Comment by michaelbrewer** [comment]
This should be a related effort to https://github.com/awslabs/aws-lambda-powertools-python/discussions/337

**Comment by michaelbrewer** [comment]
@heitorlessa hasn't this already been resolved? or should we create an issue on the backlog to resolve this?
